src/pe/export.rs

```diff
@@ -56,52 +56,79 @@ pub type ExportOrdinalTable = Vec<u16>;
+                    table.push(name_rva);
+                }
+                else {
+                    break;
+                }
+            }
+
+            table
+        });
+
+        let export_ordinal_table = utils::find_offset(export_directory_table.ordinal_table_rva as usize, sections).map(|table_offset| {
+            let mut offset = table_offset;
+            let mut table: ExportOrdinalTable = Vec::with_capacity(number_of_name_pointers);
 
-        let export_address_table_offset = utils::find_offset_or(export_directory_table.export_address_table_rva as usize, sections, &format!("Cannot map export_directory_table.export_address_table_rva ({:#x}) into offset", export_directory_table.export_address_table_rva))?;
-        let export_end = export_rva + size;
-        let offset = &mut export_address_table_offset.clone();
-        let mut export_address_table: ExportAddressTable = Vec::with_capacity(address_table_entries);
-        for _ in 0..address_table_entries {
-            let rva: u32 = bytes.gread_with(offset, scroll::LE)?;
-            if utils::is_in_range(rva as usize, export_rva, export_end) {
-                export_address_table.push(ExportAddressTableEntry::ForwarderRVA(rva));
-            } else {
-                export_address_table.push(ExportAddressTableEntry::ExportRVA(rva));
+            for _ in 0..number_of_name_pointers {
+                if let Ok(name_ordinal) = bytes.gread_with(&mut offset, scroll::LE) {
+                    table.push(name_ordinal);
+                }
+                else {
+                    break;
+                }
             }
-        }
 
-        let name_offset = utils::find_offset_or(export_directory_table.name_rva as usize, sections, &format!("Cannot map export_directory_table.name_rva ({:#x}) into offset", export_directory_table.name_rva))?;
-        debug!("ExportData.parse pointers: {:#x} ordinals: {:#x} addresses: {:#x}", name_pointer_table_offset, export_ordinal_table_offset, export_address_table_offset);
-        let name: &'a str = bytes.pread(name_offset)?;
+            table
+        });
+
+        let export_address_table = utils::find_offset(export_directory_table.export_address_table_rva as usize, sections).map(|table_offset| {
+            let mut offset = table_offset;
+            let mut table: ExportAddressTable = Vec::with_capacity(address_table_entries);
+            let export_end = export_rva + size;
+
+            for _ in 0..address_table_entries {
+                if let Ok(func_rva) = bytes.gread_with::<u32>(&mut offset, scroll::LE) {
+                    if utils::is_in_range(func_rva as usize, export_rva, export_end) {
+                        table.push(ExportAddressTableEntry::ForwarderRVA(func_rva));
+                    } else {
+                        table.push(ExportAddressTableEntry::ExportRVA(func_rva));
+                    }
+                }
+                else {
```

Comment:
Nit: else on newline (doesn't match style above)

Reply:
Sorry, I did not think of the style.